std/algorithm/iteration.d

```diff
@@ -3038,10 +3057,42 @@ if (isInputRange!RoR && isInputRange!(ElementType!RoR)
                 else
                     _current = _items.front;
 
+                static if (isBidirectional)
+                    _currentBack = _items.back.save;
+
+                static if (isBidirectional)
+                {
+                    if (_currentBack.empty)
+                    {
+                        inputEndsWithEmpty = true;
+                    }
+                }
+
                 if (_current.empty)
                 {
                     // No data in the current item - toggle to use the separator
-                    useSeparator();
+                    inputStartsWithEmpty = true;
+
+                    //A range with a single element which is empty will always
```

Comment:
I'm confused by this comment.

Reply:
I was referring to this kind of example : 
```dlang
auto r = [""].joiner("abc") //will always return ""
```
I tried to make the comment clearer.